The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26050064751/job/76584267966) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/76584267966) [(plain)](https://github.com/rust-lang/rust/commit/bb99b171632fc0fcee876aff72fbae55c532df99/checks/76584267966/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
30    |                            help: remove these parentheses
+    |
+    = note: see issue #109417 <https://github.com/rust-lang/rust/issues/109417> for more information
+    = help: add `#![feature(return_type_notation)]` to the crate attributes to enable
+    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
31 
32 error[E0107]: associated type takes 1 lifetime argument but 0 lifetime arguments were supplied
33   --> $DIR/gat-trait-path-parenthesised-args.rs:6:27


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args generic-associated-types/gat-trait-path-parenthesised-args.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/generic-associated-types/gat-trait-path-parenthesised-args" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2015"
stdout: none
--- stderr -------------------------------
error: lifetimes must be followed by `+` to form a trait object type
##[error]  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:6:29
   |
LL | fn foo<'a>(arg: Box<dyn X<Y('a) = &'a ()>>) {}
   |                             ^^
   |
help: consider adding a trait bound after the potential lifetime bound
   |
LL | fn foo<'a>(arg: Box<dyn X<Y('a + /* Trait */) = &'a ()>>) {}
   |                                +++++++++++++

error: parenthesized generic arguments cannot be used in associated type constraints
##[error]  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:6:27
   |
LL | fn foo<'a>(arg: Box<dyn X<Y('a) = &'a ()>>) {}
   |                           ^^^^^
   |
help: use angle brackets instead
   |
LL - fn foo<'a>(arg: Box<dyn X<Y('a) = &'a ()>>) {}
LL + fn foo<'a>(arg: Box<dyn X<Y<'a> = &'a ()>>) {}
   |

error: parenthesized generic arguments cannot be used in associated type constraints
##[error]  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:13:27
   |
LL | fn bar<'a>(arg: Box<dyn X<Y() = ()>>) {}
   |                           ^--
   |                            |
   |                            help: remove these parentheses
   |
   = note: see issue #109417 <https://github.com/rust-lang/rust/issues/109417> for more information
   = help: add `#![feature(return_type_notation)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0107]: associated type takes 1 lifetime argument but 0 lifetime arguments were supplied
##[error]  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:6:27
   |
LL | fn foo<'a>(arg: Box<dyn X<Y('a) = &'a ()>>) {}
   |                           ^ expected 1 lifetime argument
   |
note: associated type defined here, with 1 lifetime parameter: `'a`
  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:3:8
   |
LL |   type Y<'a>;
   |        ^ --
help: add missing lifetime argument
   |
LL | fn foo<'a>(arg: Box<dyn X<Y('_, 'a) = &'a ()>>) {}
   |                             +++

error[E0107]: associated type takes 0 generic arguments but 1 generic argument was supplied
##[error]  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:6:27
   |
LL | fn foo<'a>(arg: Box<dyn X<Y('a) = &'a ()>>) {}
   |                           ^---- help: remove the unnecessary generics
   |                           |
   |                           expected 0 generic arguments
   |
note: associated type defined here, with 0 generic parameters
---

error[E0107]: associated type takes 1 lifetime argument but 0 lifetime arguments were supplied
##[error]  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:13:27
   |
LL | fn bar<'a>(arg: Box<dyn X<Y() = ()>>) {}
   |                           ^ expected 1 lifetime argument
   |
note: associated type defined here, with 1 lifetime parameter: `'a`
  --> /checkout/tests/ui/generic-associated-types/gat-trait-path-parenthesised-args.rs:3:8
   |
LL |   type Y<'a>;
   |        ^ --
help: add missing lifetime argument
   |
LL | fn bar<'a>(arg: Box<dyn X<Y('_) = ()>>) {}
   |                             ++

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0107`.
```

</details>

--- a/compiler/rustc_ast_lowering/src/lib.rs
+++ b/compiler/rustc_ast_lowering/src/lib.rs
@@ -1197,8 +1197,6 @@ impl<'hir> LoweringContext<'_, 'hir> {
                         }
                     } else {
                         self.emit_bad_parenthesized_trait_in_assoc_ty(data);
-                        // FIXME(return_type_notation): we could issue a feature error
-                        // if the parens are empty and there's no return type.
                         self.lower_angle_bracketed_parameter_data(
                             &data.as_angle_bracketed_args(),
                             ParamMode::Explicit,

